x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/CsvTests.java

```diff
@@ -478,44 +479,46 @@ private static CsvTestsDataLoader.MultiIndexTestDataset testDatasets(LogicalPlan
         return new CsvTestsDataLoader.MultiIndexTestDataset(indexName, datasets);
     }
 
-    private static TestPhysicalOperationProviders testOperationProviders(CsvTestsDataLoader.MultiIndexTestDataset datasets)
-        throws Exception {
-        var indexResolution = loadIndexResolution(datasets);
+    private static TestPhysicalOperationProviders testOperationProviders(
+        FoldContext foldCtx,
+        CsvTestsDataLoader.MultiIndexTestDataset datasets
+    ) throws Exception {
         var indexPages = new ArrayList<TestPhysicalOperationProviders.IndexPage>();
         for (CsvTestsDataLoader.TestsDataset dataset : datasets.datasets()) {
             var testData = loadPageFromCsv(CsvTests.class.getResource("/data/" + dataset.dataFileName()), dataset.typeMapping());
             indexPages.add(new TestPhysicalOperationProviders.IndexPage(dataset.indexName(), testData.v1(), testData.v2()));
         }
-        return TestPhysicalOperationProviders.create(indexPages);
+        return TestPhysicalOperationProviders.create(foldCtx, indexPages);
     }
 
     private ActualResults executePlan(BigArrays bigArrays) throws Exception {
         LogicalPlan parsed = parser.createStatement(testCase.query);
         var testDatasets = testDatasets(parsed);
         LogicalPlan analyzed = analyzedPlan(parsed, testDatasets);
 
+        FoldContext foldCtx = FoldContext.unbounded();
```

Comment:
Maybe let's add a comment: It's fine to leave this unbounded because the corresponding integration test cases will be bounded.

Reply:
I think if we're going to set these to the small limit that's probably fine to do it in tests too.